cxx-qt-gen/src/parser/mod.rs

```diff
@@ -0,0 +1,166 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+pub mod qobject;
+
+use qobject::{find_and_merge_cxx_qt_item, find_qobjects, ParserCxxQtData};
+use syn::{token::Brace, ItemMod, Result};
+
+/// A struct representing a module block with CXX-Qt relevant Items into ParserCxxQtData
+pub struct Parser {
+    /// The module which unknown (eg CXX) blocks are stored into
+    pub module: ItemMod,
+    /// Any CXX-Qt data that needs generation later
+    pub cxx_qt_data: ParserCxxQtData,
+}
+
+impl Parser {
+    /// Constructs a Parser object from a given ItemMod block
+    pub fn from(module: ItemMod) -> Result<Self> {
+        let mut new_module = module.clone();
```

Comment:
This is taking ownership of `module` so I'm unclear what the need for cloning is?

Reply:
Aha, an older version got all confused with mutable and immutable borrows, seems this new layout doesn't need this clone. Good spot! :-)